src/mosaic-node-manager.cc

```diff
@@ -62,154 +65,214 @@ namespace ns3 {
             internet.Install(singleNode);
             NetDeviceContainer netDevices = m_wifi80211pHelper.Install(m_wifiPhyHelper, m_waveMacHelper, singleNode);
             m_ipAddressHelper.Assign(netDevices);
 
-        } else if (m_commType == LTE) {
-
-            NS_LOG_INFO ("Creating helpers for the LTE...");
-            // Associate the node with buildings for better radio propagation modeling
-            BuildingsHelper::Install(singleNode);
-
-            // Ensure that the mobility models of all nodes are consistent with their positions
-            BuildingsHelper::MakeMobilityModelConsistent(); 
-
-            // Install an LTE device on the node
-            NetDeviceContainer ueDev = m_lteHelper->InstallUeDevice(singleNode);
-            m_ueDevs.Add(ueDev);
-            // Install the internet stack on the node
-            InternetStackHelper internet;
-            internet.Install(singleNode);
+            //Install app
+            NS_LOG_INFO("Install MosaicProxyApp application on node " << singleNode->GetId());
+            Ptr<MosaicProxyApp> app = CreateObject<MosaicProxyApp>();
+            app->SetNodeManager(this);
+            singleNode->AddApplication(app);
+            app->SetRxSocket();
 
-            // Assign an IPv4 address to the LTE device
-            Ipv4InterfaceContainer vehicleIpIface = m_epcHelper->AssignUeIpv4Address(ueDev);
+            //Install mobility model
+            NS_LOG_INFO("Install MosaicMobilityModel on node " << singleNode->GetId());
+            Ptr<ConstantVelocityMobilityModel> mobModel = CreateObject<ConstantVelocityMobilityModel>();
+            mobModel->SetPosition(position);
+            singleNode->AggregateObject(mobModel);
 
-            // Set up static routing for the node to use the default gateway provided by the EPC helper
-            Ipv4StaticRoutingHelper Ipv4RoutingHelper;
-            Ptr<Ipv4StaticRouting> vehicleStaticRouting = Ipv4RoutingHelper.GetStaticRouting(singleNode->GetObject<Ipv4>());
-            vehicleStaticRouting->SetDefaultRoute(m_epcHelper->GetUeDefaultGatewayAddress(), 1);
-
-            // Attach the LTE device to the eNodeB (base station)
-            m_lteHelper->Attach(ueDev);
-
-            // Create and activate a sidelink bearer for V2X communication
-            Ptr<LteSlTft> tft = Create<LteSlTft>(LteSlTft::BIDIRECTIONAL, m_clientRespondersAddress, m_groupL2Address); 
-            m_lteV2xHelper->ActivateSidelinkBearer(Simulator::Now(), ueDev, tft);
-            m_ns3ID2UniqueAddress[ID] = m_clientRespondersAddress;
-            m_groupL2Address++;
-            m_clientRespondersAddress = Ipv4AddressGenerator::NextAddress (Ipv4Mask ("255.255.0.0"));
+        } else if (m_commType == LTE) {
 
-            // Install the V2X sidelink configuration on the LTE device
-            m_lteHelper->InstallSidelinkV2xConfiguration(ueDev, m_ueSidelinkConfiguration);            
+            m_mosaic2ns3ID[ID] = m_ueNodeIdList.front();
```

Comment:
The DSRC branch of this if statement has a check for if that node is deactivated on line 232, do we need a similar check here? Or is that a unique condition to DSRC?

Reply:
Update code to cover this part